src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Logging/OutputWindowLogger.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Razor.Workspaces.Extensions;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.Logging;
+
+[Shared]
+[Export(typeof(OutputWindowLogger))]
+internal class OutputWindowLogger : ILogger
+{
+    private const LogLevel MinimumLogLevel = LogLevel.Warning;
```

Comment:
Is this class always used, or just in tests? Is there value in presenting all of our warnings and errors to all users? I'm thinking of the cases where people have reported things like the "No configurations received" warning, which is a complete red herring.

Reply:
Always used. I would suggest that there is value in presenting all of our warnings and errors to all users to help everyone understand what's going on, and if we run into cases where people are being shown unhelpful or uninformative warnings/errors then the solution is to either prevent those supposedly bad scenarios or demote the log which is not correctly classified.